src/Jobs/Import.php

```diff
@@ -21,6 +21,8 @@ final class Import
      */
     private $source;
 
+    public int $timeout = 60;
```

Comment:
I think these timeouts need to be `?int`, and `null` by default so that the timeouts would fall back to the queue worker timeout setting, otherwise you're overriding the queue worker's timeout by default. Same applies to the other job, the config file, and these lines:

```php
$job->timeout = config('elasticsearch.queue.timeout', 60);
```

Just be mindful that env vars always come as strings, so empty timeouts of 0 should be converted to `null`s there.

Reply:
I could not be sure at this point. I changed the properties as nullable integer, but I could not find a proper place to set zero or empty valued config as `null`.

Reply:
You can do it either in the config file, or when reading from the config.

If doing it when reading from the config, you'll want to do something like:

```php
$job->timeout = (int) config('elasticsearch.queue.timeout', '60') ?: null;
```

Remember, env var values come in as strings (I think), so you need to do the conversion.

Rant: Also technically 0 is also a valid timeout to express that the queue worker should restart right away after processing a job, though I would think that a lot more often than not people set timeouts to 0 by mistake, because they think that 0 timeout means no timeout, when actually not. So I personally don't think it's important to keep the 0 number of 0 was given explicitly.

Reply:
I think there is so much logic here, and I don't want to put all of them in to config.php file. Maybe, `Matchish\ScoutElasticSearch\ElasticSearch\Config` may be the better place to implement this logic with a function. What do you think?

Reply:
Why not in ImportCommand then? That's the only place you read the config.
But in any case you can put the logic wherever you feel is right for the project.

Reply:
👍